tests/attention/test_modular_fmha_prefill_paged.py

```
@@ -0,0 +1,700 @@
+            head_dim_qk=128,
+            causal=True,
+            q_data_type=torch.bfloat16,
+            kv_data_type=torch.float8_e4m3fn,
+        )
+    # mixed K at run (only V may differ from the plan); the top-level
+    # wrapper's q/k dtype check fires first, the internal wrapper's
+    # k_cache check backstops direct callers.
+    w = _paged_wrapper()
+    w.plan(qo, kv_pg, ids, lpl, page_size=16, **plan_kw)
+    k_c, v_c = cache[:, 0].to(torch.float8_e4m3fn), cache[:, 1]
+    with pytest.raises(ValueError, match="k_cache|dtype of k"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)